src/DataValues/MatrixValue.php

```diff
@@ -0,0 +1,128 @@
+<?php
+namespace DataValues;
+
+use DataValues\DecimalValue;
+/**
+ * Class representing a matrix.
+ *
+ * @licence GNU GPL v2+
+ * @author Andrius Merkys < andrius.merkys@gmail.com >
+ */
+class MatrixValue extends DataValueObject {
+
+	/**
+	 * @var array[]
+	 */
+	private $value;
+
+	/**
+	 * Constructs a new MatrixValue object, representing the given value.
+	 *
+	 * @var array[] $value 2-dimensional array of numbers (int|float|string|DecimalValue).
+	 *
+	 * @throws IllegalValueException
+	 */
+	public function __construct( $value ) {
+		if ( !is_array( $value ) ) {
+			throw new IllegalValueException( '$value must be an array' );
+		}
+
+		if ( count( $value ) == 0 ) {
+			throw new IllegalValueException( '$value can not be an empty array' );
+		}
+
+		$value_now = array();
+
+		$columns = null;
+		foreach ( $value as $row ) {
+			if( !is_array( $row ) ) {
+				throw new IllegalValueException( '$value must be a 2-dimensional array' );
+			}
+			if( $columns == null ) {
+				$columns = count( $row );
+			} else if( $columns != count( $row ) ) {
+				throw new IllegalValueException( 'all rows of $value must be of the same length' );
+			}
+
+			/**
+			 * Filtering the matrix, converting int|float|string into
+			 * DecimalValue.
+			 */
```

Comment:
Could you please explain it? The check for the matrix input be placed in a separate method, am I right?